### PR TITLE
PHP 7.2 support

### PR DESCRIPTION
**Pull request by lackovic10**
- [x] remove usages of each()

### CODE REVIEW

**Comment by marcosdsanchez** [comment]
Thanks! Let's use this ticket to provide support for php 7.2 .

Can you please disallow failures in travis? https://github.com/ChessCom/Chess-Game/blob/develop/.travis.yml#L8-L9 .

File: Tests/ChessGameTest.php [hunk 1]
```diff
@@ -1105,4 +1105,30 @@ private function invokeMethod(&$object, $methodName, array $parameters = array()
         $method->setAccessible(true);
         return $method->invokeArgs($object, $parameters);
     }
+
+    /**
+     * @dataProvider validMoveProvider
+     */
+    public function testValidMove(array $move, bool $isValid): void
```
Comment (marcosdsanchez) [inline Tests/ChessGameTest.php]: This test passes in latest develop, so it's not really showing that this fixes the deprecation warning in PHP7.2+

Comment (lackovic10) [reply Tests/ChessGameTest.php]: i think it's because of the setup you guys have on chess.com
on chesskid, this is failing

Comment (lackovic10) [reply Tests/ChessGameTest.php]:
```
1) Chesskid\Tests\Unit\ChessGame\DataTransformer\SanMoveListToPiotrTransformerTest::testTransformStandardChess with data set #0 (Chesskid\Core\Entity\ChessGame\SanMoveList Object (...), 'kA0Kow3N')
The each() function is deprecated. This message will be suppressed on further calls

/www/sites/www.chesskid-dev.com/current/vendor/chesscom/chess-game/src/Chess/Game/ChessGame.php:2293
/www/sites/www.chesskid-dev.com/current/src/Chesskid/Core/Entity/Game/ChessGame.php:339
/www/sites/www.chesskid-dev.com/current/src/Chesskid/Core/Service/ChessGame/DataTransformer/SanMoveListToPiotrTransformer.php:62
/www/sites/www.chesskid-dev.com/current/src/Chesskid/Core/Service/ChessGame/DataTransformer/SanMoveListToPiotrTransformer.php:31
/www/sites/www.chesskid-dev.com/current/src/Chesskid/Tests/Unit/ChessGame/DataTransformer/SanMoveListToPiotrTransformerTest.php:29
/www/sites/www.chesskid-dev.com/current/src/Chesskid/Main/MainTest.php:65
/www/sites/www.chesskid-dev.com/current/src/Chesskid/Main/MainTest.php:18
```

**Comment by lackovic10** [comment]
@marcosdsanchez i've pushed https://github.com/ChessCom/Chess-Game/pull/34/commits/3b485802aa3ac6bb5bbf12cd3f49b9941256cb90, let me know if this is what you've expected for the travis configuration

File: Tests/ChessGameTest.php (hunk 1 above)
Comment (marcosdsanchez) [reply Tests/ChessGameTest.php]: The test is not needed. Develop already fails in latest develop as we can see in this link: https://travis-ci.org/ChessCom/Chess-Game/jobs/476987303

**Review by marcosdsanchez — APPROVED** [review]
Great. Please remove the test. Travis was already failing in latest develop and it's now passing for php 7.2 in this branch.

**Comment by lackovic10** [comment]
@marcosdsanchez i think the test is not redundant, it's testing the `_validMove()` method which i think is fine, but i removed the test anyway.

**Pull request merged**